### ISSUE TITLE
Cannot read vg file.

### ISSUE THREAD

**Issue by djhogan** [vgteam/vg#2382]
Please describe:

I'm trying to write a program in C++ that parses a vg graph file output by `vg construct`. I used x.vg (from the `test/small` example) as a test. 

```
../bin/vg construct -r small/x.fa -v small/x.vcf.gz >x.vg
```

I copied vg.proto from master HEAD and generated the vg.pb.{h,cc} files using protoc (v3.7.1). Here is the code for my program:

```
...
#include "vg.pb.h"

int main(int argc, char *argv[]){
	vg::Graph g;

	fstream input("x.vg", ios::in | ios::binary);
	if (input.is_open()){
		if (!g.ParseFromIstream(&input)){
			cout << "parse error" << endl;
		}
	} else {
		cout << "couldn't open the file" << endl;
	}
	
	return 0;
}
```

The program only outputs "parse error". What's going on?

**Comment by jeizenga** [comment]
@adamnovak Is this because of the `libvgio` header?

**Comment by jeizenga** [comment]
Update: I think I was correct. Adam (who would know better than me) is out on vacation right now, but you could try linking in `libvgio` from https://github.com/vgteam/libvgio. The readme on that repository shows how to load a `Graph`.

**Comment by glennhickey** [comment]
`x.vg` isn't straight protobuf, so you can't read it directly with the protobuf API.  There's an explanation and discussion in this thread #1997 

tldr: if you're creating graphs with vg, then you need to load them through the vg api (linked to above by Jordan)

**Comment by djhogan** [comment]
@jeizenga libvgio did the trick, thanks.
